crates/flox-rust-sdk/src/models/flox_package.rs

```diff
@@ -211,9 +211,44 @@ pub enum ParseError {
     NoPackage,
 }
 
+impl Display for FloxTriple {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        if self.stability != Stability::Stable {
+            write!(f, "{}.", self.stability)?;
+        }
+        if self.channel != "nixpkgs-flox" {
+            write!(f, "{}.", self.channel)?;
+        }
+        write!(f, "{}", self.name)?;
+        if let Some(version) = &self.version {
+            write!(f, "@{}", version)?;
+        }
+        Ok(())
+    }
+}
+
 impl Display for FloxPackage {
-    fn fmt(&self, _f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        todo!()
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            FloxPackage::Triple(triple) => write!(f, "{}", triple),
+            _ => todo!(),
+        }
+    }
+}
+
+pub fn packages_to_string(packages: &Vec<FloxPackage>) -> String {
+    match packages.len() {
+        0 => "".to_string(),
+        1 => format!("'{}'", packages[0]),
+        2 => format!("'{}' and '{}'", packages[0], packages[1]),
+        _ => {
+            let mut result = String::new();
+            for package in &packages[0..packages.len() - 1] {
```

Comment:
You don't _need_ to change this, but just so you know, a more idiomatic Rust way of doing this would be either 

```rust
let almost_all = packages.len() - 1;
for package in packages.iter().take(almost_all) {
...
}
```

or

```rust
let almost_all = packages.len() - 1;
packages.iter()
    .take(almost_all)
    .for_each(|p| {
        result.push_str(...);
    });
```

It can be helpful just to skim the list of methods on the `Iterator` trait to see if there's something that fits the bill. I always find methods in there that I didn't know existed (or more likely I knew at one point and forgot about).

Reply:
> ```rust
> let last = packages.pop().unwrap();
> let result = format!("{} and '{last}'", packages.join(", ");
> ```

That will result in `a, b and c` rather than `a, b, and c`

Reply:
Moved to utils::display and combined some of both of your suggestions